### PR TITLE
feat: upstream spec to ecotone

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

Update ecotone spec.

### CODE REVIEW

File: specs/protocol/derivation.md
Comment (seolaoh) [inline specs/protocol/derivation.md:1]: Please modify L118 to below
```md
block its [L1 origin][g-l1-origin]. The epoch's number equals that of its L1 origin block.
```
also L137 to below
```md
the [L2 genesis block][g-l2-genesis-block] as the first L2 block. We then derive L2 blocks from each epoch in order,
```

**Comment by seolaoh** [comment]
Why `lint-links` action not working?

**Comment by kangsorang** [comment]
> Why `lint-links` action not working?
**_-it_** option in docker run makes problem. I removed it.

File: specs/protocol/exec-engine.md [hunk 1]
```diff
@@ -123,12 +137,50 @@ can be accessed in two interchangeable ways:
...
+
+The two base fee values and their respective scalars can be accessed in two interchangeable ways:
 
-<!--
-*Note: the [Engine API][l1-api-spec] is in alpha, `v1.0.0-alpha.5`.
-There may be subtle tweaks, beta starts in a few weeks*
--->
+- read from the deposited L1 attributes (`l1BaseFeeScalar`, `l1BlobBaseFeeScalar`, `basefee`,
+  `blobBaseFee`) of the current L2 block
+- read from the L1 Block Info contract (`0x4200000000000000000000000000000000000015`)
```
Comment (Pangssu) [inline specs/protocol/exec-engine.md]:
```suggestion
- read from the L1 Block Info contract (`0x4200000000000000000000000000000000000002`)
```

File: specs/protocol/exec-engine.md [hunk 2]
```diff
@@ -123,12 +137,50 @@ can be accessed in two interchangeable ways:
...
+  - using the respective solidity getter functions
+  - using direct storage-reads:
+    - basefee `uint256` in slot `1`
+    - blobBaseFee `uint256` in slot `7`
```
Comment (Pangssu) [inline specs/protocol/exec-engine.md]:
```suggestion
    - blobBaseFee `uint256` in slot `8`
```

**Comment by seolaoh** [comment]
Which commit did you use of Optimism spec? How about adding `UPSTREAM.md` like kroma?

File: specs/glossary.md [hunk 3]
```diff
@@ -717,9 +734,14 @@ See also the [Building The Payload Attributes][building-payload-attr] section of
 
 The L2 genesis [block] is the first block of the [L2] chain in its current version.
 
-The state of the L2 genesis block contains [Predeployed contracts][predeploy].
+The state of the L2 genesis block comprises:
+
+- State inherited from the previous version of the L2 chain.
+  - This state was possibly modified by "state surgeries". For instance, the migration to Bedrock entailed changes on
+    how native ETH balances were stored in the storage trie.
+- [Predeployed contracts][predeploy]
```
Comment (seolaoh) [inline specs/glossary.md]: Please revert this

File: specs/protocol/deposits.md [hunk 4]
```diff
@@ -238,12 +242,64 @@ This transaction MUST have the following values:
...
+| blobBaseFeeScalar | uint32  | 8-11           |         |
+| sequenceNumber    | uint64  | 12-19          |         |
+| l1BlockTimestamp  | uint64  | 20-27          |         |
+| l1BlockNumber     | uint64  | 28-35          |         |
+| basefee           | uint256 | 36-67          | 2       |
+| blobBaseFee       | uint256 | 68-99          | 3       |
+| l1BlockHash       | bytes32 | 100-131        | 4       |
+| batcherHash       | bytes32 | 132-163        | 5       |
+
+Total calldata length MUST be exactly 164 bytes, implying the sixth and final segment is only
```
Comment (seolaoh) [inline specs/protocol/deposits.md]: Please change it to ours

File: specs/protocol/deposits.md [hunk 5]
```diff
@@ -306,6 +371,30 @@ After running `pnpm build` in the `packages/contracts` directory, the bytecode t
 file will be located in the `deployedBytecode` field of the build artifacts file at
 `/packages/contracts/artifacts/contracts/L2/L1Block.sol/L1Block.json`.
 
+#### Ecotone L1Block upgrade
+
+The L1 Attributes Predeployed contract, `L1Block.sol`, is upgraded as part of the Ecotone upgrade.
+The version is incremented to `1.2.0`, one new storage slot is introduced, and one existing slot
```
Comment (seolaoh) [inline specs/protocol/deposits.md]:
```suggestion
The version is incremented to `1.1.0`, one new storage slot is introduced, and one existing slot
```

File: specs/protocol/derivation.md [hunk 6]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+The `L1Block` contract is upgraded to process the new Ecotone L1-data-fee parameters and L1 blob base-fee.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4210000000000000000000000000000000000000`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `375,000`
+- `data`: `0x60806040523480156100105...` ([full bytecode](../static/bytecode/ecotone-l1-block-deployment.txt))
```
Comment (seolaoh) [inline specs/protocol/derivation.md:1137]: How about just removing `ecotone-l1-block-deployment.txt` file or fix it to the correct one? I think the bytecode is different from ours.

File: specs/protocol/derivation.md [hunk 7]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+Verify `data`:
+
+```bash
+git checkout 5996d0bc1a4721f2169ba4366a014532f31ea932
+pnpm clean && pnpm install && pnpm build
+jq -r ".bytecode.object" packages/contracts-bedrock/forge-artifacts/L1Block.sol/L1Block.json
+```
+
+This transaction MUST deploy a contract with the following code hash
+`0xc88a313aa75dc4fbf0b6850d9f9ae41e04243b7008cf3eadb29256d4a71c1dfd`.
```
Comment (seolaoh) [inline specs/protocol/derivation.md]: Also these lines are not applicable to ours.

File: specs/protocol/derivation.md [hunk 8]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+
+The `L1Block` contract is upgraded to process the new Ecotone L1-data-fee parameters and L1 blob base-fee.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4210000000000000000000000000000000000000`
+- `to`: `null`
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `375,000`
```
Comment (seolaoh) [inline specs/protocol/derivation.md]:
```suggestion
- `gasLimit`: `500,000`
```

File: specs/protocol/derivation.md [hunk 9]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+will use the blob base fee to compute the gas price for L1-data-fee transactions.
+
+A deposit transaction is derived with the following attributes:
+
+- `from`: `0x4210000000000000000000000000000000000001`
+- `to`: `null`,
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `1,000,000`
+- `data`: `0x60806040523480156100...` ([full bytecode](../static/bytecode/ecotone-gas-price-oracle-deployment.txt))
```
Comment (seolaoh) [inline specs/protocol/derivation.md:1167]: How about just removing `ecotone-gas-price-oracle-deployment.txt` file or fix it to the correct one? I think the bytecode is different from ours.

File: specs/protocol/derivation.md [hunk 10]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+Verify `data`:
+
+```bash
+git checkout 5996d0bc1a4721f2169ba4366a014532f31ea932
+pnpm clean && pnpm install && pnpm build
+jq -r ".bytecode.object" packages/contracts-bedrock/forge-artifacts/GasPriceOracle.sol/GasPriceOracle.json
+```
+
+This transaction MUST deploy a contract with the following code hash
+`0x8b71360ea773b4cfaf1ae6d2bd15464a4e1e2e360f786e475f63aeaed8da0ae5`.
```
Comment (seolaoh) [inline specs/protocol/derivation.md]: Also these lines are not applicable to ours.

File: specs/protocol/derivation.md [hunk 11]
```diff
@@ -941,6 +1098,254 @@ entries.
...
+
+- `from`: `0x0B799C86a49DEeb90402691F1041aa3AF2d3C875`, as specified in the EIP.
+- `to`: null
+- `mint`: `0`
+- `value`: `0`
+- `gasLimit`: `0x3d090`, as specified in the EIP.
+- `isCreation`: `true`
+- `data`:
+  `0x60618060095f395ff33373fffffffffffffffffffffffffffffffffffffffe14604d57602036146024575f5ffd5b5f35801560495762001fff810690815414603c575f5ffd5b62001fff01545f5260205ff35b5f5ffd5b62001fff42064281555f359062001fff015500`
+- `isSystemTx`: `false`, as per the Regolith upgrade, even the system-generated transactions spend gas.
```
Comment (seolaoh) [inline specs/protocol/derivation.md]:
```suggestion
```

File: specs/protocol/exec-engine.md [hunk 12]
```diff
@@ -264,20 +387,62 @@ the operation within the engine is the exact same as with L1 (although with an E
...
 
 [rollup node spec]: rollup-node.md
 
+## Ecotone: disable Blob-transactions
+
+[EIP-4844] introduces Blob transactions: featuring all the functionality of an [EIP-1559] transaction,
+plus a list of "blobs": "Binary Large Object", i.e. a dedicated data type for serving Data-Availability as base-layer.
+
+With the Ecotone upgrade, all Cancun L1 execution features are enabled, with [EIP-4844] as exception:
+as a L2, the OP-Stack does not serve blobs, and thus disables this new transaction type.
```
Comment (seolaoh) [inline specs/protocol/exec-engine.md]:
```suggestion
as a L2, the Kroma-network does not serve blobs, and thus disables this new transaction type.
```

File: specs/protocol/system-config.md [hunk 13]
```diff
@@ -101,10 +162,15 @@ The contained log events are filtered and processed as follows:
 - the remaining event data is opaque, encoded as ABI bytes (i.e. includes offset and length data),
   and encodes the configuration update. In version `0` the following types are supported:
   - type `0`: `batcherHash` overwrite, as `bytes32` payload.
-  - type `1`: `overhead` and `scalar` overwrite, as two packed `uint256` entries.
+  - type `1`: Pre-Ecotone, `overhead` and `scalar` overwrite, as two packed `uint256`
+    entries. After Ecotone upgrade, `overhead` is ignored and `scalar` interpreted as a [versioned
+    encoding](#ecotone-scalar-overhead-uint256uint256-change) that updates `baseFeeScalar` and
+    `blobBaseFeeScalar`.
   - type `2`: `gasLimit` overwrite, as `uint64` payload.
   - type `3`: `unsafeBlockSigner` overwrite, as `address` payload.
```
Comment (seolaoh) [inline specs/protocol/system-config.md:170]: Please add type 4, validatorRewardScalar

**Review by 0xHansLee — APPROVED** [review]

File: specs/protocol/deposits.md [hunk 14]
```diff
@@ -238,12 +242,65 @@ This transaction MUST have the following values:
...
+#### L1 Attributes - Ecotone
+
+On the Ecotone activation block, and if Ecotone is not activated at Genesis,
+the L1 Attributes Transaction includes a call to `setL1BlockValues()`
+because the L1 Attributes transaction precedes the [Ecotone Upgrade Transactions][ecotone-upgrade-txs],
+meaning that `setL1BlockValuesEcotone` is not guaranteed to exist yet.
+
+Every subsequent L1 Attributes transaction should include a call to the `setL1BlockValuesEcotone()` function.
+The input args are no longer ABI encoded function parameters,
+but are instead packed into 5 32-byte aligned segments (starting after the function selector).
```
Comment (seolaoh) [inline specs/protocol/deposits.md]:
```suggestion
but are instead packed into 6 32-byte aligned segments (starting after the function selector).
```

File: specs/protocol/deposits.md [hunk 15]
```diff
@@ -238,12 +242,65 @@ This transaction MUST have the following values:
...
+| baseFeeScalar         | uint32  | 4-7            | 1       |
+| blobBaseFeeScalar     | uint32  | 8-11           |         |
+| sequenceNumber        | uint64  | 12-19          |         |
+| l1BlockTimestamp      | uint64  | 20-27          |         |
+| l1BlockNumber         | uint64  | 28-35          |         |
+| basefee               | uint256 | 36-67          | 2       |
+| blobBaseFee           | uint256 | 68-99          | 3       |
+| l1BlockHash           | bytes32 | 100-131        | 4       |
+| batcherHash           | bytes32 | 132-163        | 5       |
+| validatorRewardScalar | uint256 | 164-195        | 6       |
```
Comment (seolaoh) [inline specs/protocol/deposits.md:286]:
```suggestion
| validatorRewardScalar | bytes32 | 164-195        | 6       |
```

File: specs/protocol/system-config.md [hunk 16]
```diff
@@ -101,9 +162,15 @@ The contained log events are filtered and processed as follows:
...
+  - type `4`: `validatorRewardScalar` overwrite, as `uint256` payload.
```
Comment (seolaoh) [inline specs/protocol/system-config.md:171]:
```suggestion
  - type `4`: `validatorRewardScalar` overwrite, as `bytes32` payload.
```

**Review by seolaoh — APPROVED** [review]
LGTM

**Review by Pangssu — APPROVED** [review]

**Pull request merged**